src/app/pages/MarginTradePage/components/NotificationSettingsDialog/index.tsx

```diff
@@ -45,6 +46,14 @@ export const NotificationSettingsDialog: React.FC<INotificationSettingsDialogPro
   const [isDiscordActive, setIsDiscordActive] = useState(false);
 
   const emailIsValid = useMemo(() => !email || validateEmail(email), [email]);
+  const isSubscribedToD1Notifications = (subscriptions: Subscription[]) => {
```

Comment:
You should be able to extract this function to outside of this component definition (so above line 26), currently it redefines new reference to this function every time component renders - not a big thing, but good practice.